transhumandesign/kag-base
# [added] Bison death animation

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

There are remaining issues, although they are minor:
- Dead Bison should have an adjusted, smaller hitbox.
- Bison should not do a death cry and a normal "boo" sound at the same time.

## Description

This PR can be considered a partial fix to #872. It improves animals. :P

This PR adds a death animation to Bison. It does this in the same fashion as when a runner/character dies, using "gib health".
When Bison's health drops to 0, FleshHit.as will manage hits to determine if gib health is exceeded; Also AnimalDeath.as was added that behaves similarly to RunnerDeath.as, tagging Bison with "dead". I took a step further and let Bison decompose if it has died but nobody has gibbed it. In that case it will turn into a skeleton.

Gib health is 6 hearts (value 3.0f).
I also did some small improvements to the code here and there.
Added the unused `BisonDie.ogg` sound and Bison can now use both "BisonBoo" and "BisonBoo2", instead of only "BisonBoo".

If a dead Bison is gibbed within 45 sec of its death, it will gib into steaks as usual.
If it takes longer, it gets tagged "decomposed" and some scripts such as "FleshHit.as" are removed.
I changed the Arrow logic a bit so Arrows will collide with Bison even when it is dead.

I introduced tag `"collide with arrows"` for Fireplace and Bison, so there aren't hard-coded checks in `Arrow.as - doesCollideWithBlob()`. 
I introduced tag "can decompose" for Bison, which is used in AnimalDeath.as.
I kept AnimalDeath.as general so it can be re-used for Shark or other Animals in the future.

I changed the Seats and Healthbar GUI/logic so seat pointers and healthbar aren't shown and you can't attach to Bison after it has died.

If you want to merge both this and #1293 , or if you want to merge only the dead state not the skeleton state, speak to me on Discord.

## Steps to Test or Reproduce

Go to Sandbox and spawn Bison.
Kill it, notice the death animation.
Keep hitting it to gib it into steaks or wait 45 sec and watch the decomposing animation.

## Video
Video - https://www.youtube.com/watch?v=iLuoOz1ZZPc
In the video you will see the dead skeleton Bison eating food. That has already been fixed in this PR.
